src/showcase/app/shared/component-viewer/stackblitz-writer/stackblitz-writer.service.ts

```diff
@@ -0,0 +1,207 @@
+ */
+const CONTENT_PATH = 'assets/docs-content/examples-source/';
+
+/**
+ * Path that refers to the generic project template files.
+ */
+const TEMPLATE_PATH = '/src/showcase/assets/stackblitz/';
+
+/**
+ * Filenames of the generic project template files.
+ */
+const TEMPLATE_FILES = [
+  'angular.json',
+  'tsconfig.json',
+  'src/main.ts',
+  'src/polyfills.ts',
+  'src/styles.css',
+  'src/app/sbb.module.ts',
+  'src/app/app.module.ts',
+];
+
+/**
+ * Stackblitz tags
+ */
+const TAGS: string[] = ['sbb', 'angular', 'example'];
+
+/**
+ * Dependencies for the Stackblitz project
+ * '*': latest version
+ */
+const dependencies = {
+  '@angular/animations': '*',
+  '@angular/cdk': '*',
+  '@angular/common': '*',
+  '@angular/compiler': '*',
+  '@angular/core': '*',
+  '@angular/forms': '*',
+  '@angular/platform-browser': '*',
+  '@angular/platform-browser-dynamic': '*',
+  '@angular/router': '*',
+  '@sbb-esta/angular-core': '*',
+  '@sbb-esta/angular-icons': '*',
+  '@sbb-esta/angular-public': '*',
+  '@sbb-esta/angular-business': '*',
+  'core-js': '2.6.9',
+  rxjs: '*',
+  tslib: '*',
+  'zone.js': '*',
+};
+
+/**
+ * StackBlitz writer, write example files to StackBlitz.
+ *
+ * StackBlitz API
+ * URL: https://run.stackblitz.com/api/aio/v1/
+ * data: {
+ *   // File name, directory and content of files
+ *   files[file-name1]: file-content1,
+ *   files[directory-name/file-name2]: file-content2,
+ *   // Can add multiple tags
+ *   tags[0]: tag-0,
+ *   // Description of StackBlitz
+ *   description: description,
+ *   // Private or not
+ *   private: true
+ *  // Dependencies
+ *  dependencies: dependencies
+ * }
+ */
+@Injectable()
+export class StackblitzWriterService {
+  constructor(private _http: HttpClient) {}
+
+  /**
+   * Returns an HTMLFormElement that will open a new StackBlitz template with the example data when
+   * called with submit().
+   */
+  constructStackBlitzForm(data: ExampleData): Promise<HTMLFormElement> {
+    const indexFile = `${data.indexFilename}.ts`;
+    const form = this._createFormElement(indexFile);
+
+    TAGS.forEach((tag, i) => this._appendFormInput(form, `tags[${i}]`, tag));
+    this._appendFormInput(form, 'private', 'true');
+    this._appendFormInput(form, 'description', data.description);
+    this._appendFormInput(form, 'dependencies', JSON.stringify(dependencies));
+
+    return new Promise((resolve) => {
+      this._addFileToForm(
+        form,
+        data,
+        `<sbb-${data.selectorName}>loading</sbb-${data.selectorName}>`,
+        'src/index.html',
+        '',
+        false
+      );
+
+      TEMPLATE_FILES.forEach((file) =>
+        this._readFile(file, TEMPLATE_PATH).subscribe(
+          (response: string) => {
+            if (file.includes('app.module.ts')) {
+              // add currently selected component to declarations and bootstrap it
+              response = response.replace(/\{0\}/g, data.componentName);
```

Comment:
I think it would be better to use the corresponding name as the token key: e.g. {componentName} instead of {0}